toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexStartupActivity.kt

```diff
@@ -5,24 +5,50 @@
 
 package org.rust.toml.crates.local
 
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.StartupActivity
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.RefreshQueue
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
 import com.intellij.openapiext.isUnitTestMode
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.ide.experiments.RsExperiments
 import org.rust.openapiext.isFeatureEnabled
+import java.io.File
+import java.nio.file.Paths
 
 /**
  * [CratesLocalIndexService] initializer.
- * Only tries to get service instance, without any updates to its crates index.
+ * Tries to get a service instance and binds crates local index update cargo registry index repository changes.
  */
 class CratesLocalIndexStartupActivity : StartupActivity.Background {
     override fun runActivity(project: Project) {
         if (isUnitTestMode) return
         if (!project.cargoProjects.hasAtLeastOneValidProject) return
 
         if (isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)) {
-            CratesLocalIndexService.getInstance()
+            val cratesService = CratesLocalIndexService.getInstance()
+
+            val userCargoRegistryIndexLocation = Paths.get(System.getProperty("user.home"), CratesLocalIndexServiceImpl.CARGO_REGISTRY_INDEX_LOCATION).toString()
+            LocalFileSystem.getInstance().addRootToWatch(userCargoRegistryIndexLocation, true)
+
+            val root = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(File(userCargoRegistryIndexLocation)) ?: return
+            VfsUtilCore.processFilesRecursively(root) { true }
+            RefreshQueue.getInstance().refresh(true, true, null, root)
+
+            ApplicationManager.getApplication().messageBus.connect().subscribe(VirtualFileManager.VFS_CHANGES, object : BulkFileListener {
+                override fun after(events: MutableList<out VFileEvent>) {
+                    if (events.any { it.path.startsWith(userCargoRegistryIndexLocation) }) {
+                        cratesService.updateIfNeeded()
+                    }
+                }
```

Comment:
Looks like we invoke this code on each project opening, i.e if a user opens two rust projects, we will set up two VFS listeners.
Yeah, we won't invoke two update tasks because of `org.rust.toml.crates.local.CratesLocalIndexServiceImpl#updateIfNeeded` implementation but anyway, let's fix it.

The solution is quite simple - just to move all this logic into init block of `CratesLocalIndexServiceImpl`